contracts/rewards/src/contract/execute.rs

```diff
@@ -0,0 +1,184 @@
+        let current_epoch = Epoch {
+            epoch_num: 0,
+            block_height_started: 0,
+            rewards: rewards_per_epoch.clone(),
+        };
+        let stored_params = StoredParams {
+            params: RewardsParams {
+                participation_threshold: (Uint64::new(1), Uint64::new(2)).try_into().unwrap(),
+                epoch_duration: epoch_duration.try_into().unwrap(),
+                rewards_per_epoch,
+            },
+            last_updated: current_epoch.clone(),
+        };
+        let mut contract = create_contract(stored_params.clone());
+
+        // epoch shouldn't change
+        let new_epoch = contract
+            .get_current_epoch(current_epoch.block_height_started + 1)
+            .unwrap();
+        assert_eq!(new_epoch, current_epoch);
+
+        // epoch should increase by one
+        let block =
+            current_epoch.block_height_started + u64::from(stored_params.params.epoch_duration);
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 1,
+                block_height_started: block,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+
+        // epoch should increase by one, but start of epoch is in the past
+        let block =
+            current_epoch.block_height_started + u64::from(stored_params.params.epoch_duration) + 1;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 1,
+                block_height_started: block - 1,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+
+        // epoch should increase by more than one
+        let block =
+            current_epoch.block_height_started + u64::from(stored_params.params.epoch_duration) * 4;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 4,
+                block_height_started: block,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+
+        // epoch should increase by more than one, but start of epoch is in the past
+        let block = current_epoch.block_height_started
+            + u64::from(stored_params.params.epoch_duration) * 4
+            + 10;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 4,
+                block_height_started: block - 10,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+    }
```

Comment:
make these different tests or find a way to parameterize them so it's easier to understand what's going on